lib/init.coffee

```diff
@@ -11,4 +20,39 @@ module.exports =
         'Available codes: https://pypi.python.org/pypi/pep257#error-codes')
 
   activate: ->
-    console.log 'activate linter-pep257'
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-pep257.executablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+    @subscriptions.add atom.config.observe 'linter-pep257.ignoreCodes',
+      (ignoreCodes) =>
+        @ignoreCodes = ignoreCodes
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    provider =
+      grammarScopes: ['source.python']
+      scope: 'file' # or 'project'
+      lintOnFly: true # must be false for scope: 'project'
+      cmd: @executablePath
+      ignoreCodes: @ignoreCodes
+      lint: (textEditor)->
+        filePath = textEditor.getPath()
+        type = 'Warning'    # all messages are of type "Warning"
+        params = [filePath]
+        if @ignoreCodes
+            params.push("--add-ignore=" + @ignoreCodes)
+        return helpers.exec(@cmd, params, {stream: 'stderr'}).then (stderr) ->
+          output = stderr.split(/\r?\n/)
+          msg = ""
+          # combine multiline messages
```

Comment:
I'm not familiar with the concept of "traces" in this context.
Is this a method of parsing the output?
Where can learn more about it?